test/NodeGuard.Tests/Rpc/GRPCAuthInterceptorTests.cs

```diff
@@ -0,0 +1,98 @@
+
+using Grpc.Core;
+using NodeGuard.Data.Models;
+using NodeGuard.Data.Repositories.Interfaces;
+using NSubstitute.ExceptionExtensions;
+
+namespace NodeGuard.Rpc;
+
+public class AuthInterceptorTests
+{
+    [Fact]
+    public async Task AuthInterceptor_NoTokenProvided()
+    {
+        // Arrange
+        var context = TestServerCallContext.Create();
+        var mockedApiTokenRepository = new Mock<IAPITokenRepository>();
+        var interceptor = new GRPCAuthInterceptor(mockedApiTokenRepository.Object);
+        var continuation = new UnaryServerMethod<string, string>(async (request, context) => { return "response"; });
+
+        var expectedMessage = "No token provided";
+        
+        // Act & Assert
+        var exception = await Assert.ThrowsAsync<RpcException>(() => interceptor.UnaryServerHandler<string, string>(String.Empty, context, continuation));
+        Assert.Equal(expectedMessage, exception.Status.Detail);
```

Comment:
You can use "WithMessage" like in this example if, I think it would be more clear what you are asserting

https://github.com/Elenpay/NodeGuard/blob/b52f42a09dad81f87e491fc9117c9d9ac06186a6/test/NodeGuard.Tests/Rpc/NodeGuardServiceTests.cs#L1300

Reply:
+1, we should use fluent assertions lib, its more easy to read.

Reply:
I am not sure this is "more clear", as the message is actually clear here:
`var expectedMessage = "Invalid token";` 
and having in a string even the inner structure of grpc internal error exceptions in my opinion is verbose since its a well defined structure by the grpc protocol... but in any case guys, i will change it just to follow same structure as the other tests!